python/lsst/pipe/drivers/skyCorrection.py

```diff
@@ -156,12 +161,30 @@ def run(self, expRef):
                     calibs = pool.mapToPrevious(self.collectSky, dataIdList)
                     makeCameraImage(camera, calibs, "sky" + extension)
 
+            exposures = self.smoothFocalPlaneSubtraction(camera, pool, dataIdList)
+
             # Persist camera-level image of calexp
             image = makeCameraImage(camera, exposures)
             expRef.put(image, "calexp_camera")
 
             pool.mapToPrevious(self.write, dataIdList)
 
+    def smoothFocalPlaneSubtraction(self, camera, pool, dataIdList):
+        '''Do 2nd Focal Plane subtraction
+
+        After doSky, we get smooth focal plane image.
+        (Before doSky, sky pistons remain in HSC-G)
+        Now make smooth focal plane background and subtract it.
+        '''
+        bgModel = FocalPlaneBackground.fromCamera(self.config.bgModel2, camera)
+        data = [Struct(dataId=dataId, bgModel=bgModel.clone()) for dataId in dataIdList]
+        bgModelList = pool.mapToPrevious(self.accumulateModel, data)
+        for ii, bg in enumerate(bgModelList):
+            self.log.info("Background %d: %d pixels", ii, bg._numbers.array.sum())
+            bgModel.merge(bg)
+        exposures = pool.mapToPrevious(self.subtractModel, dataIdList, bgModel)
```

Comment:
This is the same code as before (lines 138-143). Can you factor it into a common method so the code isn't duplicated?